backend/src/main/kotlin/com/group7/artshare/service/SearchService.kt

```diff
@@ -0,0 +1,103 @@
+package com.group7.artshare.service
+
+import com.group7.artshare.DTO.ArtItemDTO
+import com.group7.artshare.DTO.DiscussionPostDTO
+import com.group7.artshare.DTO.OnlineGalleryDTO
+import com.group7.artshare.DTO.PhysicalExhibitionDTO
+import com.group7.artshare.entity.*
+import com.group7.artshare.repository.*
+import org.springframework.stereotype.Service
+
+
+@Service
+class SearchService (
+    private val artItemRepository: ArtItemRepository,
+    private val eventInfoRepository: EventInfoRepository,
+    private val physicalExhibitionRepository: PhysicalExhibitionRepository,
+    private val onlineGalleryRepository: OnlineGalleryRepository,
+    private val registeredUserRepository: RegisteredUserRepository,
+    private val accountInfoRepository: AccountInfoRepository,
+    private val discussionPostRepository: DiscussionPostRepository,
+    private val artItemInfoRepository: ArtItemInfoRepository
+){
+
+    private fun searchArtItem(keys : String) : List<ArtItemDTO>{
+        var infoList =  artItemInfoRepository.findFullTextSearch(keys).mapNotNull { it.id }
+        return artItemRepository.findAllByArtItemInfo_IdIn(infoList).mapNotNull { it.mapToDTO() }
+    }
+
+    private fun searchPhysicalExhibition(keys:String) : List<PhysicalExhibitionDTO>{
+        var infoList = eventInfoRepository.findFullTextSearch(keys).mapNotNull { it.id }
+        return physicalExhibitionRepository.findAllByEventInfo_IdIn(infoList).mapNotNull { it.mapToDTO() }
+    }
+
+    private fun searchOnlineGallery(keys : String) : List<OnlineGalleryDTO>{
+        var infoList = eventInfoRepository.findFullTextSearch(keys).mapNotNull { it.id }
+        return onlineGalleryRepository.findAllByEventInfo_IdIn(infoList).mapNotNull { it.mapToDTO() }
+    }
+
+
+    private fun searchDiscussionPost(keys : String) : List<DiscussionPostDTO>{
+        return discussionPostRepository.findFullTextSearch(keys).mapNotNull { it.mapToDTO() }
+    }
+
+    private fun searchUser(keys : String) : List<RegisteredUser>{
```

Comment:
I think the return type should also be DTO here

Reply:
In my branch no RegisteredUserDTO exists. The changes must be done when merged to the develop branch.

Reply:
I think it exists now since you merged the develop branch. Please fix this one as well.

Reply:
Done.